src/main/scala/org/hammerlab/guacamole/assembly/DeBruijnGraph.scala

```diff
@@ -354,17 +367,103 @@ object DeBruijnGraph {
-   * @param overlapSize The amount of the sequences we expect to overlap (The number of bases the last sequence overlaps
-   *                    with the next. For a standard kmer graph, this is the length of the kmer length - 1
-   * @return A single merged sequence
-   */
+    * Merge sequences where we expect consecutive entries to overlap by `overlapSize` bases.
+    *
+    * @param sequences Set of sequences that we would like to combine into a single sequence
+    * @param overlapSize The amount of the sequences we expect to overlap (The number of bases the last sequence overlaps
+    *                    with the next. For a standard kmer graph, this is the length of the kmer length - 1
+    *
+    * @return A single merged sequence
+    */
   def mergeOverlappingSequences(sequences: Seq[Sequence], overlapSize: Int): Sequence = {
     val head = sequences.headOption.getOrElse(Seq.empty)
     val rest = sequences.tail.flatMap(sequence => sequence.takeRight(sequence.length - overlapSize + 1))
     head ++ rest
   }
 
+  /**
+   * Extract a subsequence from a read sequence
+   *
+   * NOTE: This assumes that the read entirely covers the reference region and does not check for insertions
+   * or deletion
+   *
+   * @param read Read to extract subsequence from
+   * @param startLocus Start (inclusive) locus on the reference
+   * @param endLocus End (exclusive) locus on the reference
+   * @return Subsequence overlapping [startLocus, endLocus)
+   */
+  private def getSequenceFromRead(read: MappedRead, startLocus: Int, endLocus: Int) = {
+    read.sequence.slice(startLocus - read.unclippedStart.toInt, endLocus - read.unclippedStart.toInt)
+  }
+
+  /**
+   * For a given set of reads identify all kmers that appear in the specified reference region
+   *
+   * @param reads  Set of reads to extract sequence from
+   * @param startLocus Start (inclusive) locus on the reference
+   * @param endLocus End (exclusive) locus on the reference
+   * @param minOccurrence Minimum number of times a subsequence needs to appear to be included
+   * @return List of subsequences overlapping [startLocus, endLocus) that appear at least `minOccurrence` time
+   */
+  private def getConsensusKmer(reads: Seq[MappedRead],
+                               startLocus: Int,
+                               endLocus: Int,
+                               minOccurrence: Int): Iterable[Vector[Byte]] = {
+
+    // Filter to reads that entirely cover the region
+    // Exclude reads that have any non-M Cigars (these don't have a 1 to 1 base mapping to the region)
+    val overlapping = reads
+      .filter(!_.cigarElements.exists(_.getOperator != CigarOperator.M))
+      .filter(r => r.overlapsLocus(startLocus) && r.overlapsLocus(endLocus - 1))
+
+    // Extract the sequences from each region
+    val sequences = overlapping
+      .map(r => getSequenceFromRead(r, startLocus, endLocus))
```

Comment:
OK, I see that the one use case is just here, and you've filtered out reads with non-`M` cigar-elements just before calling `getSequenceFromRead`. I would:

- inline `getSequenceFromRead`; having a glorified `MappedRead.slice` that assumes as a precondition that reads with non-`M` cigars have been filtered out seems not worth the boilerplate of making it its own method.
- `s/.filter(!…/.filterNot(…` above
- total style nit that you're free to disregard, but I prefer starting a chain of `filter`/`map`s like you have here/above with the initial object on its own line as well, so I'd rewrite this and the block above as:

```scala
    // Filter to reads that entirely cover the region.
    // Exclude reads that have any non-M Cigars (these don't have a 1-to-1 base mapping to the region).
    val sequences = 
      reads
        .filterNot(_.cigarElements.exists(_.getOperator != CigarOperator.M))
        .filter(r => r.overlapsLocus(startLocus) && r.overlapsLocus(endLocus - 1))
        .map(r => r.sequence.slice(startLocus - r.unclippedStart.toInt, endLocus - r.unclippedStart.toInt))
```

or

```scala
    // Filter to reads that entirely cover the region.
    // Exclude reads that have any non-M Cigars (these don't have a 1-to-1 base mapping to the region).
    val sequences =
      for {
        read <- reads
        if !read.cigarElements.exists(_.getOperator != CigarOperator.M)
        if read.overlapsLocus(startLocus) && read.overlapsLocus(endLocus - 1)
        unclippedStart = read.unclippedStart.toInt
      } yield {
        read.sequence.slice(startLocus - unclippedStart, endLocus - unclippedStart)
      }
```

lmk if that doesn't seem better to you